### PR TITLE
Feat/survey submission

### PR DESCRIPTION
**Pull request by edmdc**
# Pull request checklist

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current behavior

- Survey does not post to database

## New behavior

- Survey post to databasa and user style is updated in store
- Subscriptions is still processed after style survey

### CODE REVIEW

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 1]
```diff
@@ -71,6 +72,15 @@ class StyleSurvey extends PureComponent {
     const styleString = styleChoices.join(', ')
     const requestVariables = {id: this.props.user.details.id, style:styleString}
     this.props.postUserSurvey(requestVariables)
+      .then((response) => {
```
Comment (jeremiahblackol) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:73]: reminder to clean up console logs

**Review by jeremiahblackol — APPROVED** [review]

File: code/web/src/modules/crate/Item.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ class Item extends PureComponent {
 
   onClickSubscribe = (crateId) => {
     if (!this.props.user.details.style) {
+      window.localStorage.setItem('pendingSubscription', JSON.stringify(crateId))
```
Comment (emontealvo) [inline code/web/src/modules/crate/Item.js:33]: Quick way to track the desired crate subscription before survey

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 3]
```diff
@@ -2,24 +2,24 @@
...
 import { H1 } from '../../ui/typography'
-// import e from 'express'
-// import { connect } from 'react-redux'
-import { connect } from 'react-redux'
-import { Link, withRouter,  Redirect } from 'react-router-dom'
-import userRoutes from '../../setup/routes/user.js'
 
+// Actions
+import { messageShow, messageHide } from '../common/api/actions'
+import { create } from '../subscription/api/actions'
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:20]: Bring actions to display messages after survey is completed

File: code/api/src/modules/user/resolvers.js [hunk 4]
```diff
@@ -114,10 +114,11 @@ export async function update(parentValue, { id, style }) {
...
-    return await models.User.update(
+   await models.User.update(
       {
         style: userStyle
       },
       { where: { id } }
     )
-}
\ No newline at end of file
+  return models.User.findOne({where: {id}})
```
Comment (perryr16) [inline code/api/src/modules/user/resolvers.js:123]: I wrote that!

**Review by perryr16 — APPROVED** [review]

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 5]
```diff
@@ -69,8 +67,23 @@ class StyleSurvey extends PureComponent {
...
-    this.props.postUserSurvey(styleString)
+    const requestVariables = {id: this.props.user.details.id, style:styleString}
+    const crateId = this.getPendingSubscription()
+    this.props.postUserSurvey(requestVariables)
+      .then((response) => {
+        console.log(response, 'in tehn')
+        if(response.status === 200) {
+          this.props.messageShow(`Style updated to ${response.data.data.userUpdate.style}`)
+        }
+      })
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:78]: Sends survey to database

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 6]
```diff
@@ -69,8 +67,23 @@ class StyleSurvey extends PureComponent {
...
+    this.props.create({ crateId })
+      .then(response => {
+        localStorage.clear()
+        this.props.history.push(userRoutes.subscriptions.path)
+      })
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:83]: creates pending subscription

File: code/web/src/modules/styleSurvey/api/actions.js [hunk 7]
```diff
@@ -17,23 +17,28 @@ export const SURVEY_USER_REQUEST_FAILURE = 'SURVEY/USER_REQUEST_FAILURE'
...
+      fields: ['id', 'name', 'style', 'email', 'role']
     }))
       .then(response => {
-        console.log("inside dispatch in post", response)
+        const user = response.data.data.userUpdate
+        dispatch({
+          type: SET_USER,
+          user
+        })
+        return response
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/api/actions.js:38]: Updates user info in store so user only fill style survey once

File: code/web/src/modules/styleSurvey/surveyChoices.js [hunk 8]
```diff
@@ -26,7 +24,10 @@ const SurveyChoices = ({ choices, handleChoice, choiceCategory }) => {
       margin: '5px',
       padding: '3px'
     }
-    return selectionStatus ? selectedStyle : unselectedStyle
+
+    // const choice = document.getElementById(`${choiceId}`)
+
+    return false ? selectedStyle : unselectedStyle
   }
```
Comment (edmdc) [inline code/web/src/modules/styleSurvey/surveyChoices.js:31]: Could quite get styles to toggle for the selected images, hopefully is a small enough bug to fix

**Pull request merged**